src/devtools/pachdev/actions.go

```diff
@@ -0,0 +1,571 @@
+	if err != nil {
+		op.err = errors.Wrap(err, "could not load kubernetes config")
+	}
+
+	op.kubeClient, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		op.err = errors.Wrap(err, "could not create kubernetes client")
+	}
+	return op.kubeClient
+}
+
+// PachClient is an accessor for op.pachClient that lazily initializes the
+// field if it's not already set.  All access to op.kubeClient should be through
+// this method, even internally, in case it hasn't been initialized yet.
+func (op *deployOp) PachClient() *client.APIClient {
+	if op.pachClient != nil {
+		return op.pachClient
+	}
+
+	pachctlCfg := &pachctl.Config{}
+	var err error
+	op.pachClient, err = pachctlCfg.NewOnUserMachine(op.Ctx(), false)
+	if err != nil {
+		op.err = errors.Wrap(err, "could not create Pachyderm client")
+	}
+	return op.pachClient
+}
+
+// KindProvider is an accessor for op.kindProvider that lazily initializes the
+// field if it's not already set.  All access to op.kindProvider should be through
+// this method, even internally, in case it hasn't been initialized yet.
+func (op *deployOp) KindProvider() *cluster.Provider {
+	if op.kindProvider != nil {
+		return op.kindProvider
+	}
+	var kindLogger log.Logger = log.NoopLogger{}
+	if verbose {
+		// TODO(msteffen) literally no idea what good UX is here. Ideally we
+		// would log to stdout instead of stderr, but I can't access
+		// kind/pgk/internal/cli to change the logging destination. Am I holding
+		// this wrong?
+		kindLogger = cmd.NewLogger()
+	}
+	op.kindProvider = cluster.NewProvider(
+		cluster.ProviderWithLogger(kindLogger),
+		cluster.ProviderWithDocker(), // TODO(msteffen): is this necessary or appropriate?
+	)
+	return op.kindProvider
+}
+
+// helmChartSource is a helper for K8sManifest that returns the location of the
+// Pachyderm helm chart it should use.
+func (op *deployOp) helmChartSource() string {
+	if op.pachVersion == "latest" {
+		// // TODO(msteffen): this codepath has no automated tests. Maybe set up
+		// // mock GitHub? (has worked in the past, and I bet the releases endpoint
+		// // is simple)
+		// client := github.NewClient(nil)
+		// // Not sure if this is the correct call -- want to filter out alphas,
+		// // nightlies, etc.
+		// release, _, err := client.Repositories.GetLatestRelease(context.Background(), "pachyderm", "pachyderm")
+		// if err != nil {
+		// 	panic(err)
+		// }
+		// return fmt.Sprintf("https://github.com/pachyderm/helmchart/releases/download/pachyderm-%s/pachyderm-%s.tgz", *release.TagName, *release.TagName)
+		panic("not implemented")
+	}
+	if op.pachVersion == "nightly" {
+		// look up from dockerhub or something, somehow
+		panic("not implemented")
+	}
+	if op.pachVersion == "local" {
+		return "./etc/helm/pachyderm"
+	}
+	panic("do not know how to get helm chart for pachVersion " + op.pachVersion)
+}
+
+// getHelmArgs is a helper for K8sManifest that returns the arguments it should
+// pass to helm
+func (op *deployOp) getHelmArgs() []string {
```

Comment:
I have always wanted to depend on helm and not shell out to it, but this is fine.  The tests also shell out to helm in a weird way (I think the library we use bundles helm, but invokes it by passing whatever helms os.Args would be...)